### ISSUE TITLE
macos app: add wake & sleep handler

### ISSUE THREAD

**Issue by iwanbk** [threefoldtech/myceliumflut#69]
The `ping` become failed when the app put to sleep. (It is expected).

**app state**
![image](https://github.com/user-attachments/assets/17189b12-a63d-49d6-9340-4fb5e80933a8)

**ping side**
![image](https://github.com/user-attachments/assets/1787a6d3-2efa-4d19-81df-d54b664621fe)

But the `ping`  didn't instantly recovered after the app wake up again (not expected)
**app state**
![image](https://github.com/user-attachments/assets/a00e434b-bbe5-48d4-92ed-56cae7fe6e10)

**ping side**
![image](https://github.com/user-attachments/assets/957a1798-6729-469a-b636-d7972cd807d5)

The recovery time varied, sometimes almost instantly but sometimes more than 10 minutes.
The long recovery was noticed when the battery is low (< 20%)

One of the quick solution is to restart the mycelium after wake up.

**Comment by iwanbk** [comment]
> One of the quick solution is to restart the mycelium after wake up.

I have tried these things:

1. restart mycelium from within the PacketTunnelProvider, but didn't work.
Some observation:
- Stopping the tunnel doesnt tearing down the TUN device
- Starting the tunnel failed to get TUN device. We got errno 9 ( EBADF: Bad file descriptor)

2. Add sleep and wake up handler to macos NSWindow
- it still doesn't work, we don't get wake & sleep event notification
- why we do this? the only difference with current `restart` is that current restart is triggered from the NSWindow, not within PacketTunnelProvider

**Comment by iwanbk** [comment]
i found that restarting mycelium outside the PacketTunnelProvider works perfectly.
I use https://pub.dev/packages/flutter_desktop_sleep to detect `wake` event and do the restart.

**Comment by iwanbk** [comment]
I also found that the problem might be in `mycelium` itself because:
- got this error on macos side : "Frame error from TCP 192.168.1.6:55080 <-> 68.183.228.64:9651: Connection reset by peer (os error 54)"
- it already connected to the public node
```console
2024-10-02T07:28:31.439602Z  INFO mycelium::peer_manager: Accepted new inbound peer
2024-10-02T07:28:31.439650Z  INFO add_peer: mycelium::peer_manager: Added new peer peer.endpoint=Tcp [::ffff:36.80.99.115]:55229
2024-10-02T07:28:31.464380Z  INFO mycelium::router: Acquired route subnet=525:c933:ef2e:bfe7::/64 peer="TCP [::ffff:68.183.228.64]:9651 <-> [::ffff:36.80.99.115]:55229"
```
- the connection recovered consistently much faster on my own public node 
```console
024-10-02T07:30:38.961313Z  INFO mycelium::router: Acquired route subnet=525:c933:ef2e:bfe7::/64 peer="TCP 192.168.0.108:53200 <-> 68.183.228.64:9651"
```
it only take ~2 minutes versus > 10 minutes on mycelium public node

**Comment by iwanbk** [comment]
fixed at #71
